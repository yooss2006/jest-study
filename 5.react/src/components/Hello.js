import React from "react";

const Hello = ({ user }) => {
  return user?.name ? <div>hello</div> : <button>login</button>;
};

export default Hello;
