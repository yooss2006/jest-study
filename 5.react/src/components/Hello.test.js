import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Hello from "./Hello";
const user = {
  name: "jinvvv",
  age: 30,
};
const user2 = {
  age: 30,
};

//스냅 샷
test("snapshot:name 있음", () => {
  const element = renderer.create(<Hello user={user} />).toJSON();
  expect(element).toMatchSnapshot();
});
test("snapshot:name 없음", () => {
  const element = renderer.create(<Hello />).toJSON();
  expect(element).toMatchSnapshot();
});

test("hello라는 글자가 포함되었는가?", () => {
  render(<Hello user={user} />);
  const helloEl = screen.getByText(/Hello/i);
  expect(helloEl).toBeInTheDocument();
});
