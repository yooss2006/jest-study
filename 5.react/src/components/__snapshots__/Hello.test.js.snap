// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot:name 없음 1`] = `
<button>
  login
</button>
`;

exports[`snapshot:name 있음 1`] = `
<div>
  hello
</div>
`;
